misc_test.go

```diff
@@ -104,3 +105,10 @@ func TestRetryable(t *testing.T) {
 		}
 	}
 }
+
+func TestCanUseErrorsAs(t *testing.T) {
+	_ = errors.As(errors.New("error"), &RateLimitedError{})
```

Comment:
Do you want to check that we actually can match errors too? I get that we wanted to catch the panic - but probably could also check we get back true for the right sort of error maybe? Not a big thing.